src/include/platform/ConfigurationManager.h

```diff
@@ -196,6 +201,8 @@ class ConfigurationManager
     ConfigurationManager(const ConfigurationManager &)  = delete;
     ConfigurationManager(const ConfigurationManager &&) = delete;
     ConfigurationManager & operator=(const ConfigurationManager &) = delete;
+
+    CommissionableDataProvider * mCommissionableDataProvider = nullptr;
```

Comment:
The code size overhead is very small, since this is called few places. Overall the issue with SetXXXProvider/GetXXXProvider is that it creates MORE dependencies, not less, on singletons.

Reply:
Imho, having one or several singletons is not a big difference - we still don't allow multiple stack instances after all, and having several approaches for the same problem in the code doesn't help developers. Btw. maybe it doesn't make sense for `GetCommissionableDataProvider` to be virtual if there's a corresponding setter in the API.

Reply:
Will fix as a follow-up